### ISSUE TITLE
[Bug]: segfault when setting colorN variable

### ISSUE THREAD

**Issue by progfolio** [brndnmtthws/conky#1370]
### What happened?

Conky segfaults when setting any of the colorN variables in the config section.

### Version

conky 1.17.0_pre compiled 2023-01-14 for Linux

### Which OS/distro are you seeing the problem on?

Arch Linux

#### segfaults
```lua
conky.config = {
	out_to_x = false,
 	own_window = false,
 	out_to_console = true,
 	background = false,
 	total_run_times = 1,
        color0 = 'FCCCCC'
};

conky.text = [[test]]
```

#### does not segfault
```lua
conky.config = {
	out_to_x = false,
 	own_window = false,
 	out_to_console = true,
 	background = false,
 	total_run_times = 1,
        -- color0 = 'FCCCCC'
};

conky.text = [[test]]
```

### Stack trace

```GDB
Program received signal SIGSEGV, Segmentation fault.
0x00005555555723fc in get_x11_color (name=0x7fffffffdb90 "FCCCCC") at /home/n/.cache/yay/conky-git/src/conky/src/colours.cc:142
142       if (XParseColor(display, DefaultColormap(display, screen), name, &color) ==
(ins)(gdb) bt full
#0  0x00005555555723fc in get_x11_color (name=0x7fffffffdb90 "FCCCCC")
    at /home/n/.cache/yay/conky-git/src/conky/src/colours.cc:142
        color = {pixel = 0, red = 0, green = 0, blue = 0, flags = 0 '\000', pad = 0 '\000'}
#1  0x00005555555724e8 in get_x11_color (colour="FCCCCC")
    at /home/n/.cache/yay/conky-git/src/conky/src/colours.cc:169
No locals.
#2  0x00005555555cac7a in priv::colour_traits::convert (index=-2, l=...)
    at /home/n/.cache/yay/conky-git/src/conky/src/gui.h:110
No locals.
#3  conky::simple_config_setting<unsigned long, priv::colour_traits>::do_convert (
    this=<optimized out>, l=..., index=-2)
    at /home/n/.cache/yay/conky-git/src/conky/src/setting.hh:291
No locals.
#4  0x00005555555cc99e in conky::simple_config_setting<unsigned long, priv::colour_traits>::lua_setter (init=true, l=..., this=0x555555745400 <color>)
    at /home/n/.cache/yay/conky-git/src/conky/src/setting.hh:304
        s = <optimized out>
        ok = <optimized out>
        s = <optimized out>
        ok = <optimized out>
#5  priv::colour_setting::lua_setter (this=0x555555745400 <color>, l=..., 
    init=<optimized out>) at /home/n/.cache/yay/conky-git/src/conky/src/gui.cc:110
        s = {L = 0x555555763e60, n = 4}
#6  0x00005555555ad8a7 in conky::priv::config_setting_base::process_setting (l=..., 
    init=<optimized out>) at /home/n/.cache/yay/conky-git/src/conky/src/setting.cc:196
        s = {L = 0x555555763e60, n = 3}
        ptr = <optimized out>
#7  0x00005555555ae4bf in conky::set_config_settings (l=...)
    at /home/n/.cache/yay/conky-git/src/conky/src/setting.cc:283
        i = 0x555555745400 <color>
        __for_range = std::vector of length 129, capacity 129 = {
          0x555555744a00 <display_name[abi:cxx11]>, 0x555555744980 <out_to_x>, 
          0x555555744940 <use_xft>, 0x555555745000 <font>, 
          0x555555744c80 <font_template[abi:cxx11]>, 
          0x555555744cd8 <font_template[abi:cxx11]+88>, 
          0x555555744d30 <font_template[abi:cxx11]+176>, 
          0x555555744d88 <font_template[abi:cxx11]+264>, 
          0x555555744de0 <font_template[abi:cxx11]+352>, 
          0x555555744e38 <font_template[abi:cxx11]+440>, 
          0x555555744e90 <font_template[abi:cxx11]+528>, 
          0x555555744ee8 <font_template[abi:cxx11]+616>, 
          0x555555744f40 <font_template[abi:cxx11]+704>, 
          0x555555744f98 <font_template[abi:cxx11]+792>, 0x555555745400 <color>, 
          0x555555745440 <color+64>, 0x555555745480 <color+128>, 0x5555557454c0 <color+192>, 
          0x555555745500 <color+256>, 0x555555745540 <color+320>, 
          0x555555745580 <color+384>, 0x5555557455c0 <color+448>, 
          0x555555745600 <color+512>, 0x555555745640 <color+576>, 
          0x5555557453c0 <default_color>, 0x555555745380 <default_shade_color>, 
          0x555555745340 <default_outline_color>, 0x555555745300 <border_inner_margin>, 
          0x5555557452c0 <border_outer_margin>, 0x555555745280 <border_width>, 
          0x555555745680 <text_alignment>, 0x555555745240 <set_transparent>, 
          0x5555557451e0 <own_window_class[abi:cxx11]>, 
          0x555555745180 <own_window_title[abi:cxx11]>, 0x555555745140 <own_window_type>, 
          0x555555745100 <own_window_hints>, 0x5555557450c0 <own_window_argb_value>, 
          0x5555557448c0 <use_argb_visual>, 0x555555745ba0 <background_colour>, 
          0x555555745080 <own_window>, 0x555555744880 <use_xdbe>, 
          0x555555744820 <imlib_cache_size>, 
          0x555555745d60 <(anonymous namespace)::draw_blended>, 0x5555556053e0 <no_buffers>, 
          0x555555605380 <bar_fill[abi:cxx11]>, 0x555555605320 <bar_unfill[abi:cxx11]>, 
          0x5555556052c0 <github_token[abi:cxx11]>, 0x555555606800 <disable_auto_reload>, 
          0x5555556067a0 <use_spacer>, 0x555555606760 <short_units>, 
          0x555555606720 <format_human_readable>, 0x555555606040 <units_spacer[abi:cxx11]>, 
          0x555555606000 <out_to_stdout>, 0x555555605fc0 <out_to_stderr>, 
          0x555555605f40 <update_interval>, 0x555555605ee0 <update_interval_on_battery>, 
          0x555555605e80 <detect_battery[abi:cxx11]>, 
          0x555555605e40 <music_player_interval>, 0x5555556066c0 <max_text_width>, 
          0x555555606680 <stuff_in_uppercase>, 0x555555606640 <stuff_in_lowercase>, 
          0x5555556065e0 <total_run_times>, 0x5555556065a0 <fork_to_background>, 
          0x555555605960 <cpu_avg_samples>, 0x555555605920 <net_avg_samples>, 
          0x5555556058e0 <diskio_avg_samples>, 0x5555556058a0 <show_graph_scale>, 
          0x555555605860 <show_graph_range>, 0x555555606560 <graph_gradient_mode>, 
          0x5555556057a0 <gap_x>, 0x555555605760 <gap_y>, 0x555555605720 <draw_borders>, 
          0x5555556056e0 <draw_graph_borders>, 0x5555556056a0 <stippled_borders>, 
          0x555555605660 <draw_shades>, 0x555555605620 <draw_outline>, 
          0x5555556055c0 <minimum_height>, 0x555555605580 <minimum_width>, 
          0x555555605540 <maximum_width>, 0x555555605500 <utf8_mode>, 
          0x5555556054c0 <max_user_text>, 0x555555605480 <text_buffer_size>, 
          0x555555606520 <pad_percents>, 0x555555610360 <current_mail_spool>, 
          0x555555610420 <(anonymous namespace)::imap>, 
          0x5555556103c0 <(anonymous namespace)::pop3>, 0x55555573e020 <if_up_strictness>, 
          0x55555573e060 <(anonymous namespace)::_template>, 
          0x55555573e0b8 <(anonymous namespace)::_template+88>, 
          0x55555573e110 <(anonymous namespace)::_template+176>, 
          0x55555573e168 <(anonymous namespace)::_template+264>, 
          0x55555573e1c0 <(anonymous namespace)::_template+352>, 
          0x55555573e218 <(anonymous namespace)::_template+440>, 
          0x55555573e270 <(anonymous namespace)::_template+528>, 
          0x55555573e2c8 <(anonymous namespace)::_template+616>, 
          0x55555573e320 <(anonymous namespace)::_template+704>, 
          0x55555573e378 <(anonymous namespace)::_template+792>, 
          0x55555573e740 <(anonymous namespace)::default_bar_width>, 
          0x55555573e700 <(anonymous namespace)::default_bar_height>, 
          0x55555573e6c0 <(anonymous namespace)::default_graph_width>, 
          0x55555573e680 <(anonymous namespace)::default_graph_height>, 
          0x55555573e640 <(anonymous namespace)::default_gauge_width>, 
          0x55555573e600 <(anonymous namespace)::default_gauge_height>, 
          0x55555573e5c0 <(anonymous namespace)::store_graph_data_explicitly>, 
          0x55555573e560 <(anonymous namespace)::console_graph_ticks>, 
          0x55555573e800 <output_unit>, 0x55555573e840 <times_in_seconds>, 
          0x55555573e8e0 <top_name_width>, 0x55555573e8a0 <top_name_verbose>, 
          0x55555573fa40 <extra_newline>, 0x55555573fb20 <overwrite_file>, 
          0x55555573fac0 <append_file>, 0x55555573fca0 <(anonymous namespace)::xftalpha>, 
          0x55555573ff80 <(anonymous namespace)::lua_load>, 
          0x55555573ff20 <(anonymous namespace)::lua_startup_hook>, 
          0x55555573fec0 <(anonymous namespace)::lua_shutdown_hook>, 
          0x55555573fe60 <(anonymous namespace)::lua_draw_hook_pre>, 
          0x55555573fe00 <(anonymous namespace)::lua_draw_hook_post>, 
          0x555555744600 <top_cpu_separate>, 
          0x555555744780 <(anonymous namespace)::mpd_port>, 
          0x555555744720 <(anonymous namespace)::mpd_host>, 
          0x5555557446c0 <(anonymous namespace)::mpd_password>, 
          0x5555557447c0 <max_port_monitor_connections>, 0x5555557449c0 <head_index>, 
          0x555555744900 <forced_redraw>, 0x555555745cc0 <hddtemp_host>, 
          0x555555745c60 <hddtemp_port>, 
          0x555555745da0 <(anonymous namespace)::imlib_cache_flush_interval>, 
          0x555555745fc0 <out_to_ncurses>}
        __for_begin = <optimized out>
        __for_end = <optimized out>
        v = std::vector of length 129, capacity 129 = {
          0x555555744a00 <display_name[abi:cxx11]>, 0x555555744980 <out_to_x>, 
          0x555555744940 <use_xft>, 0x555555745000 <font>, 
          0x555555744c80 <font_template[abi:cxx11]>, 
          0x555555744cd8 <font_template[abi:cxx11]+88>, 
          0x555555744d30 <font_template[abi:cxx11]+176>, 
          0x555555744d88 <font_template[abi:cxx11]+264>, 
          0x555555744de0 <font_template[abi:cxx11]+352>, 
          0x555555744e38 <font_template[abi:cxx11]+440>, 
          0x555555744e90 <font_template[abi:cxx11]+528>, 
          0x555555744ee8 <font_template[abi:cxx11]+616>, 
          0x555555744f40 <font_template[abi:cxx11]+704>, 
          0x555555744f98 <font_template[abi:cxx11]+792>, 0x555555745400 <color>, 
          0x555555745440 <color+64>, 0x555555745480 <color+128>, 0x5555557454c0 <color+192>, 
          0x555555745500 <color+256>, 0x555555745540 <color+320>, 
          0x555555745580 <color+384>, 0x5555557455c0 <color+448>, 
          0x555555745600 <color+512>, 0x555555745640 <color+576>, 
          0x5555557453c0 <default_color>, 0x555555745380 <default_shade_color>, 
          0x555555745340 <default_outline_color>, 0x555555745300 <border_inner_margin>, 
          0x5555557452c0 <border_outer_margin>, 0x555555745280 <border_width>, 
          0x555555745680 <text_alignment>, 0x555555745240 <set_transparent>, 
          0x5555557451e0 <own_window_class[abi:cxx11]>, 
          0x555555745180 <own_window_title[abi:cxx11]>, 0x555555745140 <own_window_type>, 
          0x555555745100 <own_window_hints>, 0x5555557450c0 <own_window_argb_value>, 
          0x5555557448c0 <use_argb_visual>, 0x555555745ba0 <background_colour>, 
          0x555555745080 <own_window>, 0x555555744880 <use_xdbe>, 
          0x555555744820 <imlib_cache_size>, 
          0x555555745d60 <(anonymous namespace)::draw_blended>, 0x5555556053e0 <no_buffers>, 
          0x555555605380 <bar_fill[abi:cxx11]>, 0x555555605320 <bar_unfill[abi:cxx11]>, 
          0x5555556052c0 <github_token[abi:cxx11]>, 0x555555606800 <disable_auto_reload>, 
          0x5555556067a0 <use_spacer>, 0x555555606760 <short_units>, 
          0x555555606720 <format_human_readable>, 0x555555606040 <units_spacer[abi:cxx11]>, 
          0x555555606000 <out_to_stdout>, 0x555555605fc0 <out_to_stderr>, 
          0x555555605f40 <update_interval>, 0x555555605ee0 <update_interval_on_battery>, 
          0x555555605e80 <detect_battery[abi:cxx11]>, 
          0x555555605e40 <music_player_interval>, 0x5555556066c0 <max_text_width>, 
          0x555555606680 <stuff_in_uppercase>, 0x555555606640 <stuff_in_lowercase>, 
          0x5555556065e0 <total_run_times>, 0x5555556065a0 <fork_to_background>, 
          0x555555605960 <cpu_avg_samples>, 0x555555605920 <net_avg_samples>, 
          0x5555556058e0 <diskio_avg_samples>, 0x5555556058a0 <show_graph_scale>, 
          0x555555605860 <show_graph_range>, 0x555555606560 <graph_gradient_mode>, 
          0x5555556057a0 <gap_x>, 0x555555605760 <gap_y>, 0x555555605720 <draw_borders>, 
          0x5555556056e0 <draw_graph_borders>, 0x5555556056a0 <stippled_borders>, 
          0x555555605660 <draw_shades>, 0x555555605620 <draw_outline>, 
          0x5555556055c0 <minimum_height>, 0x555555605580 <minimum_width>, 
          0x555555605540 <maximum_width>, 0x555555605500 <utf8_mode>, 
          0x5555556054c0 <max_user_text>, 0x555555605480 <text_buffer_size>, 
          0x555555606520 <pad_percents>, 0x555555610360 <current_mail_spool>, 
          0x555555610420 <(anonymous namespace)::imap>, 
          0x5555556103c0 <(anonymous namespace)::pop3>, 0x55555573e020 <if_up_strictness>, 
          0x55555573e060 <(anonymous namespace)::_template>, 
          0x55555573e0b8 <(anonymous namespace)::_template+88>, 
          0x55555573e110 <(anonymous namespace)::_template+176>, 
          0x55555573e168 <(anonymous namespace)::_template+264>, 
          0x55555573e1c0 <(anonymous namespace)::_template+352>, 
          0x55555573e218 <(anonymous namespace)::_template+440>, 
          0x55555573e270 <(anonymous namespace)::_template+528>, 
          0x55555573e2c8 <(anonymous namespace)::_template+616>, 
          0x55555573e320 <(anonymous namespace)::_template+704>, 
          0x55555573e378 <(anonymous namespace)::_template+792>, 
          0x55555573e740 <(anonymous namespace)::default_bar_width>, 
          0x55555573e700 <(anonymous namespace)::default_bar_height>, 
          0x55555573e6c0 <(anonymous namespace)::default_graph_width>, 
          0x55555573e680 <(anonymous namespace)::default_graph_height>, 
          0x55555573e640 <(anonymous namespace)::default_gauge_width>, 
          0x55555573e600 <(anonymous namespace)::default_gauge_height>, 
          0x55555573e5c0 <(anonymous namespace)::store_graph_data_explicitly>, 
          0x55555573e560 <(anonymous namespace)::console_graph_ticks>, 
          0x55555573e800 <output_unit>, 0x55555573e840 <times_in_seconds>, 
          0x55555573e8e0 <top_name_width>, 0x55555573e8a0 <top_name_verbose>, 
          0x55555573fa40 <extra_newline>, 0x55555573fb20 <overwrite_file>, 
          0x55555573fac0 <append_file>, 0x55555573fca0 <(anonymous namespace)::xftalpha>, 
          0x55555573ff80 <(anonymous namespace)::lua_load>, 
          0x55555573ff20 <(anonymous namespace)::lua_startup_hook>, 
          0x55555573fec0 <(anonymous namespace)::lua_shutdown_hook>, 
          0x55555573fe60 <(anonymous namespace)::lua_draw_hook_pre>, 
          0x55555573fe00 <(anonymous namespace)::lua_draw_hook_post>, 
          0x555555744600 <top_cpu_separate>, 
          0x555555744780 <(anonymous namespace)::mpd_port>, 
          0x555555744720 <(anonymous namespace)::mpd_host>, 
          0x5555557446c0 <(anonymous namespace)::mpd_password>, 
          0x5555557447c0 <max_port_monitor_connections>, 0x5555557449c0 <head_index>, 
          0x555555744900 <forced_redraw>, 0x555555745cc0 <hddtemp_host>, 
          0x555555745c60 <hddtemp_port>, 
          0x555555745da0 <(anonymous namespace)::imlib_cache_flush_interval>, 
          0x555555745fc0 <out_to_ncurses>}
        s = {L = <optimized out>, n = 0}
#8  0x000055555557b97c in initialisation (argc=argc@entry=3, argv=argv@entry=0x7fffffffdf88)
    at /usr/include/c++/12.2.0/bits/unique_ptr.h:191
        act = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, sa_mask = {
            __val = {0 <repeats 16 times>}}, sa_flags = 0, sa_restorer = 0x0}
        oact = {__sigaction_handler = {sa_handler = 0x0, sa_sigaction = 0x0}, sa_mask = {
            __val = {0 <repeats 16 times>}}, sa_flags = 0, sa_restorer = 0x0}
#9  0x000055555556b4f5 in main (argc=3, argv=0x7fffffffdf88)
    at /home/n/.cache/yay/conky-git/src/conky/src/main.cc:351
No locals.
```

**Comment by bi4k8** [comment]
This should be fixed by PR #1390.

**Comment by progfolio** [comment]
no longer an issue
